src/components/admin/form-components/withSelector.jsx

```diff
@@ -0,0 +1,177 @@
+          `The ${this.props.mode} ${title} has already been added.`,
+        );
+        return;
+      }
+      const newObjects = update(this.props.elements, {
+        $push: [{ id, name: title, slug }],
+      });
+      this.props.onUpdate(newObjects);
+    };
+
+    handleDelete(id, title, slug) {
+      // disabled this if saving
+      if (this.props.disabled) return;
+
+      const objectSlug = !slug ? this.props.slugify(title) : slug;
+
+      const index = this.props.elements.findIndex(
+        baseObject =>
+          !('slug' in baseObject)
+            ? this.props.slugify(baseObject.name) === objectSlug
+            : baseObject.slug === objectSlug,
+      );
+      if (index === -1) {
+        throw new Error(
+          'You tried to delete an object not currently selected. ' +
+            "This shouldn't happen, please let the developers know that it did.",
+        );
+      }
+      const newObjects = update(this.props.elements, { $splice: [[index, 1]] });
+      this.props.onUpdate(newObjects);
+    }
+
+    render() {
+      const styles = {
+        wrapper: {
+          display: 'flex',
+          flexWrap: 'wrap',
+        },
+      };
+
+      const objectChips =
+        this.props.elements.length > 0
+          ? _.map(this.props.elements, object => (
+              <ObjectChip
+                key={object.id !== null ? object.id : object.name}
+                id={object.id}
+                title={object.name}
+                onDelete={this.handleDelete}
+                style={{ margin: 4 }}
+              >
+                {object.name}
+              </ObjectChip>
+            ))
+          : null;
+
+      const noObjectsMessage = (
+        <span style={{ color: 'rgba(0, 0, 0, 0.3)' }}>
+          No {this.props.mode} are currently assigned to this article
+        </span>
+      );
+
+      return (
+        <div>
+          <br />
+          <p style={{ marginTop: 0, marginBottom: 10 }}>
+            {capFirstLetter(this.props.mode)}
+          </p>
+          <div style={styles.wrapper}>{objectChips || noObjectsMessage}</div>
+          <WrappedField
+            length={3}
+            handleClick={this.handleClickAdd}
+            {...this.props}
+          />
+        </div>
+      );
+    }
+  }
+
+  Selector.propTypes = {
+    elements: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.number,
+        name: PropTypes.string.isRequired,
+      }),
+    ).isRequired,
+    onChange: PropTypes.func.isRequired,
+    onUpdate: PropTypes.func.isRequired,
+    displayAlert: PropTypes.func.isRequired,
+    slugify: PropTypes.func.isRequired,
+    // It isn't a typo and they are required in SearchBar so no need for defaults
+    /* eslint-disable react/no-typos, react/require-default-props */
+    mode: React.PropTypes.oneOf(['staff', 'articles']).isRequired,
+    /* eslint-enable react/no-typos, react/require-default-props */
+    disabled: PropTypes.bool,
+    enableAdd: PropTypes.bool,
```

Comment:
We don't use enableAdd anywhere anymore right? So that just goes directly to the `WrappedField` right? Are we sure that we want to enforce which options the WrappedField should take at this level? (May be, just double checking), because I think maybe this component is flexible enough that you could just use the `...WrappedField.propTypes` pattern and then it would automatically typecheck for whatever component you put into the HOC

Reply:
@emilgoldsmith I agree that that would be the easiest way. In a comment on a different PR, though, you said that you'd like to have them visible, so that it is easier for developers to reference the documentation, instead of them having to go other files.

Reply:
Well for 1, wasn't my point the same as now? To use the spread syntax to make it clear that those props were also necessary. Here I'm just saying that you're limiting the implementation of the components that this HOC can be used on as they have to use `enableAdd` etc. I'm not 100% sure whether that limitation is accurate or not, could be, though my gut tells me no, I think you'll have a better idea of that.

Making it variable tells the developer that he'll need to supply all the props he would normally have supplied to the wrapped component (also, you could use the `_.omit` lodash function to show that you've already provided length (and I think you also hardcoded another prop in right?)

Do I make sense?

Reply:
Yep, you do :)! And you were right. I forgot that! Sorry. 

In terms of omitting length, I want it to be overridable instead. I will make it a default component of searchBar :)

Reply:
In the updated version I completely ignored it as it is added by default in the Searchbar component.